platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingPersistenceManager.java

```diff
@@ -191,21 +210,39 @@ private Object decryptIfNecessary(String key, Object value) {
         throw new IllegalStateException(initFailureMessage);
       }
 
-      synchronized (EncryptionAgent.class) {
-        if (!Arrays.asList(children).contains("meta")) {
-          KeyczarTool.main(
-              new String[] {
-                "create", "--location=" + pwDirString, "--purpose=crypt", "--name=Password"
-              });
-          KeyczarTool.main(
-              new String[] {"addkey", "--location=" + pwDirString, "--status=primary"});
-        }
+      String keysetLocation = Paths.get(pwDirString, KEYSET_FILE_NAME).toString();
+
+      synchronized (EncryptingPersistenceManager.class) {
+        File keysetFile = new File(keysetLocation);
+        InputStream keysetFileInputStream = null;
+        OutputStream keysetFileOutputStream = null;
         try {
-          this.crypter = new Crypter(pwDirString);
-        } catch (Exception e) {
+          AeadConfig.register();
+          if (!keysetFile.exists()) {
+            keysetHandle = KeysetHandle.generateNew(AeadKeyTemplates.AES128_GCM);
+            keysetFileOutputStream = Files.newOutputStream(Paths.get(keysetLocation));
+            CleartextKeysetHandle.write(
+                keysetHandle, JsonKeysetWriter.withOutputStream(keysetFileOutputStream));
+          } else {
+            keysetFileInputStream = Files.newInputStream(Paths.get(keysetLocation));
+            keysetHandle =
+                CleartextKeysetHandle.read(JsonKeysetReader.withInputStream(keysetFileInputStream));
+          }
+          aead = AeadFactory.getPrimitive(keysetHandle);
+        } catch (GeneralSecurityException | IOException e) {
           initFailureMessage = "Could not setup encryption. ";
           initException = e;
           throw new IllegalStateException(initFailureMessage, initException);
+        } finally {
+          // close streams
+          try {
+            keysetFileInputStream.close();
+          } catch (IOException | NullPointerException ignore) {
```

Comment:
✏️ Do a null check instead of catching the NPE. Can a try with resources be used?